shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -540,165 +365,34 @@ void TextInputChannel::EnterPressed(TextInputModel* model, bool select) {
-  ecore_imf_context_retrieve_surrounding_callback_set(
-      imf_context_, RetrieveSurroundingCallback, this);
-
-  // These APIs have to be set when IMF's setting status is changed.
-  ecore_imf_context_autocapital_type_set(imf_context_,
-                                         ECORE_IMF_AUTOCAPITAL_TYPE_NONE);
-  ecore_imf_context_prediction_allow_set(imf_context_, EINA_FALSE);
-  ecore_imf_context_input_panel_layout_set(imf_context_,
-                                           ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL);
-  ecore_imf_context_input_panel_return_key_type_set(
-      imf_context_, ECORE_IMF_INPUT_PANEL_RETURN_KEY_TYPE_DEFAULT);
-  ecore_imf_context_input_panel_layout_variation_set(imf_context_, 0);
-  ecore_imf_context_input_panel_language_set(
-      imf_context_, ECORE_IMF_INPUT_PANEL_LANG_AUTOMATIC);
-}
-
-void TextInputChannel::UnregisterIMFCallback() {
-  ecore_imf_context_event_callback_del(imf_context_, ECORE_IMF_CALLBACK_COMMIT,
-                                       CommitCallback);
-  ecore_imf_context_event_callback_del(
-      imf_context_, ECORE_IMF_CALLBACK_PREEDIT_CHANGED, PreeditCallback);
-  ecore_imf_context_event_callback_del(imf_context_,
-                                       ECORE_IMF_CALLBACK_DELETE_SURROUNDING,
-                                       DeleteSurroundingCallback);
-  ecore_imf_context_event_callback_del(imf_context_,
-                                       ECORE_IMF_CALLBACK_PRIVATE_COMMAND_SEND,
-                                       PrivateCommandCallback);
-  ecore_imf_context_input_panel_event_callback_del(
-      imf_context_, ECORE_IMF_INPUT_PANEL_STATE_EVENT,
-      InputPanelStateChangedCallback);
-  ecore_imf_context_input_panel_event_callback_del(
-      imf_context_, ECORE_IMF_INPUT_PANEL_GEOMETRY_EVENT,
-      InputPanelGeometryChangedCallback);
-}
-
 void TextInputChannel::ConsumeLastPreedit() {
-  if (have_preedit_) {
+  if (text_editing_context_.has_preedit_) {
     std::string before = active_model_->GetText();
-    int count = preedit_end_pos_ - preedit_start_pos_;
+    int count = text_editing_context_.preedit_end_pos_ -
+                text_editing_context_.preedit_start_pos_;
     active_model_->DeleteSurrounding(-count, count);
     std::string after = active_model_->GetText();
     FT_LOGI("Consume last preedit count:[%d] text:[%s] -> [%s]", count,
             before.data(), after.data());
     SendStateUpdate(*active_model_);
   }
-  have_preedit_ = false;
-  preedit_end_pos_ = 0;
-  preedit_start_pos_ = 0;
+  text_editing_context_.has_preedit_ = false;
+  text_editing_context_.preedit_end_pos_ = 0;
+  text_editing_context_.preedit_start_pos_ = 0;
 }
 
-void TextInputChannel::ResetCurrentContext() {
-  SetEditStatus(EditStatus::kNone);
-  ecore_imf_context_reset(imf_context_);
-  preedit_start_pos_ = 0;
-  preedit_end_pos_ = 0;
-  have_preedit_ = false;
+bool TextInputChannel::ShouldNotUseFilterEvent(std::string key, bool is_ime) {
```

Comment:
Maybe `ShouldNotFilterEvent`?

Actually I think `HandleEvent` is more appropriate than `FilterEvent` but I think you followed Ecore's terminology.

Reply:
That's right. I followed Ecore's terminology.